src/SFML/Graphics/Shader.cpp

```diff
@@ -148,11 +164,19 @@ sfShader* sfShader_createFromStream(sfInputStream* vertexShaderStream, sfInputSt
         }
         else
         {
-            // vertex + geometry + fragment shaders
-            CallbackStream vertexStream(vertexShaderStream);
             CallbackStream geometryStream(geometryShaderStream);
-            CallbackStream fragmentStream(fragmentShaderStream);
-            success = shader->This.loadFromStream(vertexStream, geometryStream, fragmentStream);
+            if (!vertexShaderStream && !fragmentShaderStream)
+            {
+                // geometry shader only
+                success = shader->This.loadFromStream(geometryStream, sf::Shader::Geometry);
+            }
+            else
```

Comment:
What about the situation where either vertexShaderFilename or fragmentShaderFilename is not set, i.e. shouldn't there be three cases as above?